plugins/git4idea/src/git4idea/branch/GitBranchUtil.java

```diff
@@ -282,7 +283,12 @@ public static String getDisplayableBranchText(@NotNull GitRepository repository)
     }
 
     GitBranch branch = repository.getCurrentBranch();
-    String branchName = (branch == null ? "" : branch.getName());
+    String branchName;
+    if (branch instanceof GitSymbolicRef) {
+      branchName = (branch == null ? "" : ((GitSymbolicRef)branch).getPrintableName());
```

Comment:
Looks reasonable to propagate `getPresentableName()` (which I feel is a better name for the method `getPrintableName()`) to `GitRef`. Default implementation would delegate to `getName()`, while `GitSymbolicRef` would provide an alternative presentation.

And there will be one less instanceof&cast ;)